File: BACKEND/src/main/java/com/ROOMIFY/Roomify/dto/ai/AIRoomRequest.java
package com.ROOMIFY.Roomify.dto.ai;

import lombok.Data;
import java.util.List;

@Data
public class AIRoomRequest {
    private String sourceImageUrl;
    private List<String> furniture;
    private String style;
    private Double roomWidth;
    private Double roomLength;
    private Double ceilingHeight;
    private String additionalInstructions;
}

File: FRONTEND/shared/src/commonMain/kotlin/org/com/model/AIRoomRequest.kt
package org.com.model

import kotlinx.serialization.Serializable

@Serializable
data class AIRoomRequest(
    val sourceImageUrl: String,
    val furniture: List<String>,
    val style: String,
    val roomWidth: Double? = null,
    val roomLength: Double? = null,
    val ceilingHeight: Double? = null,
    val additionalInstructions: String? = null
)
